src/common/set.c

```diff
@@ -1802,6 +1804,17 @@ util_part_open(struct pool_set_part *part, size_t minsize, int create)
 			return -1;
 		}
 
+		if (Fallocate_at_create && create_part) {
+			ssize_t actual_size = util_file_get_size(part->path);
```

Comment:
util_file_get_size can fail and you are not checking for that.
What's wrong with using "size" returned by util_file_open?